drivers/gles3/storage/texture_storage.cpp

```diff
@@ -1551,11 +1551,17 @@ void TextureStorage::_update_render_target(RenderTarget *rt) {
 	}
 
 	Config *config = Config::get_singleton();
-
-	rt->color_internal_format = rt->is_transparent ? GL_RGBA8 : GL_RGB10_A2;
-	rt->color_format = GL_RGBA;
-	rt->color_type = rt->is_transparent ? GL_UNSIGNED_BYTE : GL_UNSIGNED_INT_2_10_10_10_REV;
-	rt->image_format = Image::FORMAT_RGBA8;
+	if (rt->viewport_mode == RS::VIEWPORT_MODE_3D && config->framebuffer_float_supported) {
```

Comment:
I think this is wrong, we should never be using GL_RGB16F on mobile devices.

Actually @clayjohn thats an interesting question, are you using `GL_RGB16F` for rendering in OpenGL? Seeing we're targeting low end I don't think we should.  We should always stick to `GL_RGB10_A2` and do the same luminance multiplier workaround we've done on the mobile Vulkan renderer. Or if we do want to support `GL_RGB16F`  on desktop, make this is project setting of the OpenGL renderer.

Reply:
I think the intent was to handle that with `framebuffer_float_supported` but the implementation may be problematic in the case you descrided.

Reply:
Yeah lets see what Clay has to say about it. It's definitely not something that is switched on transparency and the needed changes to run on `GL_RGB10_A2` may be far more complex than what we're looking into right now.

Reply:
Just reviewed the code and discussion we had with @clayjohn on the previous proposal (#61667), i think the `framebuffer_float_supported` should do the job on mobile rendering. For desktop (with float allowed) i tried on my setup and the float value are working well (similar to vulkan version).

Reply:
The GL Compatibility renderer should always render to a 32bpp sRGB render target. The GL Compatibility assumes that rendering always takes place in sRGB space, for older mobile devices this is a very important optimization. In order to keep results consistent between devices, we stick with the limitations of the lowest common denominator. That way users can develop on desktop and expect that the results look the same on mobile

Reply:
I thought that checking for support of float framebuffer was done to force the sRGB on mobile with this limitation. My main usecase of the GL Compatibility is for Web so i did not test any mobile build.